services/brig/test/integration/API/User/Auth.hs

```diff
@@ -361,6 +368,28 @@ testSendLoginCode brig = do
   let _timeout = fromLoginCodeTimeout <$> responseJsonMaybe rsp2
   liftIO $ assertEqual "timeout" (Just (Code.Timeout 600)) _timeout
 
+testLoginVerify6DigitEmailCode :: Brig -> Galley -> DB.ClientState -> Http ()
+testLoginVerify6DigitEmailCode brig galley db = do
+  (u, tid) <- createUserWithTeam' brig
+  let checkLogin body status = login brig body PersistentCookie !!! const status === statusCode
+  let checkLoginFails body = checkLogin body 403
+  let checkLoginSucceeds body = checkLogin body 200
+
+  Util.setTeamSndFactorPasswordChallenge galley tid Public.TeamFeatureEnabled
+  let Just email = userEmail u
+  Util.generateVerificationCode brig (Public.SendVerificationCode Public.Login email)
+  gen <- Code.mk6DigitGen (Code.ForEmail email)
+  Just vcode <- lookupCode db gen Code.AccountLogin
+
+  let loginMissingCode = PasswordLogin (LoginByEmail email) defPassword (Just defCookieLabel) Nothing
+  checkLoginFails loginMissingCode
+
+  let loginWithCode = PasswordLogin (LoginByEmail email) defPassword (Just defCookieLabel) (Just $ Code.codeValue vcode)
+  checkLoginSucceeds loginWithCode
+
+lookupCode :: MonadIO m => DB.ClientState -> Code.Gen -> Code.Scope -> m (Maybe Code.Code)
+lookupCode db gen = liftIO . DB.runClient db . Code.lookup (Code.genKey gen)
```

Comment:
This shouldn't take a `Gen`, but only a key.  And isn't there a function that can convert an email into a key without having to go through generating a `Gen` and then throwing away the bogus code?